tools/shared/src/test/scala/org/scalajs/core/tools/linker/AnalyzerTest.scala

```diff
@@ -0,0 +1,240 @@
+package org.scalajs.core.tools.linker
+
+import org.junit.Test
+import org.junit.Assert._
+
+import org.scalajs.core.ir
+import org.scalajs.core.ir.ClassKind
+import org.scalajs.core.ir.Definitions._
+import org.scalajs.core.ir.Trees._
+import org.scalajs.core.ir.Types._
+
+import org.scalajs.core.tools.logging.NullLogger
+import org.scalajs.core.tools.io._
+import org.scalajs.core.tools.linker._
+import org.scalajs.core.tools.linker.analyzer._
+import org.scalajs.core.tools.linker.standard._
+
+import Analysis._
+
+import org.scalajs.core.tools.linker.testutils._
+
+class AnalyzerTest {
+  private val reqsFactory = SymbolRequirement.factory("unit test")
+
+  private val emptyOptHints: OptimizerHints = OptimizerHints.empty
+
+  implicit private val noPosition: ir.Position = ir.Position.NoPosition
+
+  private val fromAnalyzer = FromCore("analyzer")
+  private val fromUnitTest = FromCore("unit test")
+
+  private def trivialCtor(enclosingClassName: String) = {
+    MethodDef(static = false, Ident("init___"), Nil, NoType,
+        Some(ApplyStatically(This()(ClassType(enclosingClassName)),
+            ClassType(ObjectClass), Ident("init___"), Nil)(NoType)))(
+        emptyOptHints, None)
+  }
+
+  @Test
+  def trivialOK(): Unit = {
+    val analysis = computeAnalysis(Nil)
+    assertNoError(analysis)
+  }
+
+  @Test
+  def missingJavaLangObject(): Unit = {
+    val analysis = computeAnalysis(Nil, stdlib = false)
+    assertHasErrors(analysis,
+        MissingJavaLangObjectClass(fromAnalyzer))
+  }
+
+  @Test
+  def invalidJavaLangObject(): Unit = {
+    val invalidJLObjectDefs = Seq(
+      ClassDef(Ident(ObjectClass), ClassKind.Class, Some(Ident("Lparent")),
+          Nil, None, Nil, Nil)(emptyOptHints),
+      ClassDef(Ident(ObjectClass), ClassKind.ModuleClass, None, Nil, None,
+          Nil, Nil)(emptyOptHints),
+      ClassDef(Ident(ObjectClass), ClassKind.Class, None,
+          List(Ident("Lparent")), None, Nil, Nil)(emptyOptHints)
```

Comment:
Could you add comments on what is wrong with them and/or add named parameters for the things you set?

The named parameter thing applies in general. It is kind of hard to understand what all the `Nil` and `None` are. Maybe define a helper to create `ClassDef` that has all these optional?